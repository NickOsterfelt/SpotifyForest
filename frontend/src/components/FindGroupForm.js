import React from "react";
import { Form, Col, Button, Row } from "react-bootstrap"

/**
 * FindGroupForm is a form for searching for groups to join
 * 
 * The search box for searching groups by name is only visible if 
 * the radio button to search by name is selected
 */

function FindGroupForm({ formData, handleChange, handleSubmit, showSearchBox }) {
    //Search box input
    const searchBox = (
        <Form.Group as={Row}>
            <Col sm={10}>
                <Form.Control
                    onChange={handleChange}
                    type="text"
                    name="search"
                    value={formData.search}
                    placeholder="group name"
                    key={"formSearchBox"}
                />
            </Col>
        </Form.Group>
    )
        
    return (
        <Form onSubmit={handleSubmit}>
            <Form.Group as={Row}>
                <Col sm={10}>
                    <Form.Check
                        onChange={handleChange}
                        type="radio"
                        label="Search groups with matching top tracks"
                        name="searchType"
                        id="searchType1"
                        value="track"
                    />
                    <Form.Check
                        onChange={handleChange}
                        type="radio"
                        label="Search groups with matching top artists"
                        name="searchType"
                        id="searchType2"
                        value="artist"
                    />
                    <Form.Check
                        onChange={handleChange}
                        type="radio"
                        label="Search by group name"
                        name="searchType"
                        id="searchType3"
                        value="name"
                    />

                </Col>
            </Form.Group>
            {showSearchBox ? searchBox : ""}
            <Form.Group as={Row}>
                <Col sm={{ span: 10 }}>
                    <Button type="submit" variant={"success"}>Search Groups</Button>
                </Col>
            </Form.Group>
        </Form>
    )
}

export default FindGroupForm;