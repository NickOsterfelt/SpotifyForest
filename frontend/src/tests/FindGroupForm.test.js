import React from "react";
import { render } from "@testing-library/react";
import FindGroupForm from "../components/FindGroupForm";

const handleChange = () => {};
const handleSubmit = () => {};
it("renders without crashing", function() {
  render(<FindGroupForm 
    handleSubmit={handleSubmit}
    handleChange={handleChange} 
    formData={{ search: "" }}/>);
});

it("matches snapshot", function() {
  const { asFragment } = render(<FindGroupForm 
    handleSubmit={handleSubmit}
    handleChange={handleChange} 
    formData={{ search: "" }}/>);
  expect(asFragment()).toMatchSnapshot();
});
