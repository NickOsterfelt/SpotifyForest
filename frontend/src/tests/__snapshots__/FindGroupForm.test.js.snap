// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <div
        class="col-sm-10"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="searchType1"
            name="searchType"
            type="radio"
            value="track"
          />
          <label
            class="form-check-label"
            for="searchType1"
            title=""
          >
            Search groups with matching top tracks
          </label>
        </div>
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="searchType2"
            name="searchType"
            type="radio"
            value="artist"
          />
          <label
            class="form-check-label"
            for="searchType2"
            title=""
          >
            Search groups with matching top artists
          </label>
        </div>
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            id="searchType3"
            name="searchType"
            type="radio"
            value="name"
          />
          <label
            class="form-check-label"
            for="searchType3"
            title=""
          >
            Search by group name
          </label>
        </div>
      </div>
    </div>
    <div
      class="form-group row"
    >
      <div
        class="col-sm-10"
      >
        <button
          class="btn btn-success"
          type="submit"
        >
          Search Groups
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
